pkg/util/util.go

```diff
@@ -5,24 +5,28 @@ import (
 	"fmt"
 	"io/ioutil"
 	"net/http"
+
+	hversion "github.com/hashicorp/go-version"
 )
 
 type githubversion struct {
 	TagName string `json:"tag_name"`
 }
 
 func GetRequest(baseURL, url string) ([]byte, error) {
-	response, err := http.Get(fmt.Sprintf("%v%v", baseURL, url))
+	response, err := http.Get(fmt.Sprintf("%s%s", baseURL, url))
 	if err != nil {
 		return []byte(""), err
 	}
 	defer response.Body.Close()
-
-	data, err := ioutil.ReadAll(response.Body)
-	if err != nil {
-		return []byte(""), err
+	if response.StatusCode == 200 {
+		data, err := ioutil.ReadAll(response.Body)
+		if err != nil {
+			return []byte(""), err
+		}
+		return data, nil
 	}
-	return data, nil
+	return []byte(""), fmt.Errorf("something goes wrong, Got status code %v while sending request to %s", response.StatusCode, fmt.Sprintf("%s%s", baseURL, url))
```

Comment:
```suggestion
	return []byte(""), fmt.Errorf("something went wrong. Received status code [%v] while sending a request to [%s]", response.StatusCode, fmt.Sprintf("%s%s", baseURL, url))
```

Reply:
Let's also move the `fmt.Sprintf()` part to the first line of this function and just use it where needed...